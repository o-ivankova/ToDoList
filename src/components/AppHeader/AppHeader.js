import React from "react";
import styles from "./AppHeader.module.scss";

const AppHeader = () => {
  return <h1 className={styles.appHeader} data-testid='app-header'>TO DO</h1>;
};

export default AppHeader;
