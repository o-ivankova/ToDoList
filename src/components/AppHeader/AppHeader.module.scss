$textcolor: #785c6f;

.appHeader {
    align-items: flex-end;
    margin: 10px;
    color: $textcolor;
  }
  