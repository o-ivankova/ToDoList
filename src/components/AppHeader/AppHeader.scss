$textcolor: #785c6f;

.app-header {
    align-items: flex-end;
    margin: 10px;
    color: $textcolor;
  }
  