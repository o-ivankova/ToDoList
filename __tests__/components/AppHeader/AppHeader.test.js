import React from "react";
import '@testing-library/jest-dom';
import AppHeader from "../../../src/components/AppHeader/AppHeader";
import { expect, test, describe } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

describe("AppHeader tests", () => {
  test("component is rendered", () => {
    render(
        <AppHeader />
    );

    expect(screen.getByTestId("app-header")).not.toBeNull;
  });

  test("component has text TO DO", () => {
    render(
        <AppHeader />
    );

    expect(screen.getByTestId("app-header")).toHaveTextContent("TO DO");
  });

  test("snapshot test", () => {
    const domTree = renderer
      .create(
          <AppHeader />
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });
});