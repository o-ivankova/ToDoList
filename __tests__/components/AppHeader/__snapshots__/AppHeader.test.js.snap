// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader tests snapshot test 1`] = `
<h1
  className="appHeader"
  data-testid="app-header"
>
  TO DO
</h1>
`;
